twake/backend/node/src/core/platform/services/websocket/index.ts

```diff
@@ -33,21 +35,20 @@ export default class WebSocket extends TwakeService<WebSocketAPI> {
       io: this.service.getIo(),
     });
 
-    /**
-     * This implementation is for php old code to push on new socket.io server
-     */
-    fastify.post("/private/pusher", {}, (request, reply) => {
-      const token = (request.headers.authorization || "").trim().split("Token ").pop();
-      const secret = this.configuration.get<string>("php_pusher_secret", "");
-      if (secret && token === secret) {
-        const body = request.body as { room: string; data: any };
-        const room = body.room;
-        const data = body.data;
-        this.service.getIo().to(room).emit("realtime:event", { name: room, data: data });
-        reply.send({});
-      } else {
-        reply.code(401).send({});
-      }
+    phpnode.register((internalServer, _, next) => {
```

Comment:
Are you sure that you need to do it this side?
Why not having this code in the phpnode service? The phpnode requires websocket, and then you can expose the REST API from it, and consume the websocket service to publish event to the room

Reply:
Done